tests/unit_tests/bittensor_tests/utils/test_tokenizer_utils.py

```diff
@@ -0,0 +1,290 @@
+
+        with torch.no_grad():
+            token_batch = input_batch['input_ids'].to(device)
+            pre_model_output = server_model(input_ids=tokens['input_ids'].to(device),
+                                            attention_mask=tokens['attention_mask'].to(device),
+                                            output_hidden_states=True)
+            pre_logits = pre_model_output.logits
+
+        original_loss = get_loss_fct(pre_logits.cpu(), tokens['input_ids'])
+        enc_pre_logits = encode_forward_response_tensor(pre_logits, topk=topk).cpu()
+
+    dec_pre_logits = decode_forward_response_tensor(enc_pre_logits, tokenizer.vocab_size, topk=topk)
+    encoded_loss = get_loss_fct(dec_pre_logits, tokens['input_ids'])
+
+    # ============================================
+    # ==== Server-side: Tokenizer translation ====
+    # ============================================
+
+    with torch.no_grad():
+        probs_std = translate_logits_to_probs_std(dec_pre_logits.cpu(),
+                                                  tokens['offset_mapping'], tokens['offset_mapping_std'],
+                                                  tokenizer, std_tokenizer,
+                                                  split_map_cache, to_translation_map, from_translation_map,
+                                                  tokens['input_ids'].cpu(), token_batch.cpu(),
+                                                  skip_equivalent=False)
+
+    logits_std = torch.log(probs_std + EPSILON)
+    translated_loss = get_loss_fct(logits_std, token_batch.cpu())
+
+    return original_loss, encoded_loss, translated_loss, enc_pre_logits
+
+
+def test_tokenizer_translation():
+    r"""
+    Unit test for tokenizer translation.
+
+        Returns:
+            Asserts that tokenizer translation produces previous encoded and translated losses.
+    """
+    test_pairs = [('English-1', 'EleutherAI/gpt-j-6B', 95),
+                  ('English-1', 'benjamin/gerpt2-large', 95),
+                  ('German-1', 'benjamin/gerpt2-large', 172)]
+
+    encodings_cache_file = "tests/unit_tests/bittensor_tests/utils/test_tokenizer_utils.pt"
+
+    try:
+        encodings = torch.load(encodings_cache_file)
+
+    except FileNotFoundError as e:
+        print('FileNotFoundError: Server model results not yet saved to', encodings_cache_file)
+        print('Will first run server models (requires CUDA)...')
+
+        # === Run server models to obtain encoded logits ===
+        encodings = {}
+        for text_name, model_name, max_length in test_pairs:
+            result = tokenizer_translation(sample_text[text_name], model_name, max_length, topk=128)
+            original_loss, encoded_loss, translated_loss, enc_pre_logits = result
+            encodings[(text_name, model_name)] = (encoded_loss, translated_loss, enc_pre_logits)
+
+            print(text_name, model_name, original_loss, encoded_loss, translated_loss)
```

Comment:
I don't think we should be handling this exception here; the test should just fail if we don't have cached embeddings, especially since they're in VCS.

When we run in circleCI we don't have the resources to compute these, and it leaves the test just hanging silently as it computes the embeddings. I think we should have this code in a separate script if we ever want to change what we're doing in this test